src/xenia/kernel/xam/xam_net.cc

```diff
@@ -205,8 +206,29 @@ typedef struct {
   uint32_t probe_replies_sent_count;
 } XNQOSLISTENSTATS;
 
+typedef struct {
+  xe::be<uint32_t> dwStructSize;
+  char* lpszScheme;
+  xe::be<uint32_t> dwSchemeLength;
+  xe::be<uint32_t> nScheme;
+  char* lpszHostName;
+  xe::be<uint32_t> dwHostNameLength;
+  xe::be<uint32_t> nPort;
+  char* lpszUserName;
+  xe::be<uint32_t> dwUserNameLength;
+  char* lpszPassword;
+  xe::be<uint32_t> dwPasswordLength;
+  char* lpszUrlPath;
+  xe::be<uint32_t> dwUrlPathLength;
+  char* lpszExtraInfo;
+  xe::be<uint32_t> dwExtraInfoLength;
+} URL_COMPONENTS, *PURL_COMPONENTS;
+
 XNetStartupParams xnet_startup_params{};
 
+// Should be changed
+HINTERNET* hInternet;
```

Comment:
global pointer? We don't do that here, but I understand kinda why you did it that way